typegen/src/typegen/validate_settings.rs

```diff
@@ -0,0 +1,98 @@
+use scale_info::{form::PortableForm, PortableRegistry};
+
+use crate::TypeGenerator;
+
+use super::{error::SettingsValidationError, settings::substitutes::TryIntoSynPath};
+
+impl<'a> TypeGenerator<'a> {
```

Comment:
It feels sortof weird to me to have a separate impl block in a different place from the type (ie I wouldn't spot this function very easily if reading through the code).

If you think that a separate file is a good way to split up this logic, then perhaps you could make this a freestanding function and then add a method to the impl block next to the `TypeGenerator` definition? That way it's easy to see in one place all of the methods available.

Reply:
I think you are right, since we want to be very specific about what type paths to check in subxt, maybe this general validation function is not the right approach. :)
About the splitting into different files: It's just my personal style influenced by how you often define extension classes in C# or Dart. But It is probably better to merge it all into one `impl` block for people reading the code.